python/ray/tune/tests/test_tune_save_restore.py

```diff
@@ -0,0 +1,140 @@
+# coding: utf-8
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import shutil
+import tempfile
+import unittest
+
+import ray
+from ray import tune
+from ray.rllib import _register_all
```

Comment:
Ideally, these tests would not depend on rllib at all. It would be great if you could create a mock trainable class that implements `_save` and `_restore`, and used that class to test these. Example of such class can be found for example here: https://github.com/ray-project/ray/blob/dca1c25d88b54dbf8a6cfc22d2a2f00930309161/python/ray/tune/tests/test_trial_runner.py#L617